### PR TITLE
Somatic variantS filter loqus RD archive

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

Filter option for local archive (loqus RD) for somatic variants.

Screenshot 2023-08-15 at 11 34 54
Screenshot 2023-08-15 at 11 39 30

Adding a few RD loqus archive tags to the cancer demo case, we see them on the variantS list,
Screenshot 2023-08-15 at 15 52 48
And filtering them works:
Screenshot 2023-08-15 at 15 53 30

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

**Comment by dnil** [comment]
Fails on duplication; this is part left over from the design, part intentional making the filter dialogs more similar between different variant types and tracks. It might soon be possible to merge them again, and just have a few exceptions, like `if sv` or `if cancer`, but not this PR.

**Comment by dnil** [comment]
This should be ready for review. I noticed we are lacking an observation column on the cancer SV page, but the same is true for RD. 🙄 I made an issue for this not to complicate this PR: #4040.

**Comment by northwestwitch** [comment]
> This should be ready for review. I noticed we are lacking an observation column on the cancer SV page, but the same is true for RD. 🙄 I made a separate issue for that not to complicate this PR: #4040.

Nice, I'll review tomorrow!

File: scout/server/blueprints/variants/templates/variants/utils.html [hunk 1]
```diff
@@ -714,10 +717,21 @@
         <div class="col-2">
           {{ wtf.form_field(form.cytoband_end) }}
         </div>
+      </div>
+      <div class="row" style="margin-top:10px;">
+        <div class="col-2"></div>
         <div class="col-2">
-          {{ wtf.form_field(form.chrom_pos) }}
+          <a class="btn btn-secondary btn-sm form-control" data-bs-toggle="collapse" href="#stringcoords" aria-expanded="false" aria-controls="stringcoords">
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/templates/variants/utils.html:724]: Much nicer with the collapsed field! 👍🏻 

image

File: scout/server/blueprints/variants/templates/variants/cancer-variants.html [hunk 2]
```diff
@@ -43,7 +43,7 @@
       </div>
       <!--Expand filters form if filters were used in a previous POST request or the previous POST request form was not validated-->
       <div class="card-body panel-collapse collapse {{ 'show' if expand_search is sameas true }}" id="collapseFilters" >
-         {{ cancer_filters(form, institute, case) }}
+        <div class="card-body">{{ cancer_filters(form, institute, case) }}</div>
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/templates/variants/cancer-variants.html:46]: Why a card body inside another card body?

Comment (northwestwitch) [inline scout/server/blueprints/variants/templates/variants/cancer-variants.html:46]:
```suggestion
        <div>{{ cancer_filters(form, institute, case) }}</div>
```

**Review by northwestwitch — APPROVED** [review]
Looks good! I didn't test it, except for the new filter looks, but you did. Just a small thing!

File: scout/server/blueprints/variants/templates/variants/cancer-variants.html (hunk 2 above)
Comment (dnil) [reply scout/server/blueprints/variants/templates/variants/cancer-variants.html:46]: Ah, good catch - I just added it per template from the other variants. Without it, there are no margins for the buttons, but quite agree. I'll see if I can't fix it for all instead - perhaps with some explicit margins.

Comment (dnil) [reply scout/server/blueprints/variants/templates/variants/cancer-variants.html]: Ok, it was a little messy, but the main reason it didn't look good was a couple of missing `</div>` from the string coordinates collapse. The browser kind of solved it based on the hint from an extra body class, but not without. Reverted the others as well.

**Comment by dnil** [comment]
Thanks for the review! I tested it locally. The somatic stage cases don't really have all the annotations. The latest prod ones do have more, and presumably there are some dev v13 ones with all. I'll see if I can't make a little hybrid thing to load on stage, or convince the balsamic crew to load a recent one to see it all integrated, but we don't need to wait with merging for that. It's an added feature, and we can check that no fixes are needed once the pipeline dev features are rolled out to stage and prod.

**Pull request merged**